packages/payment-processor/src/payment/erc20-escrow-payment.ts

```diff
@@ -26,10 +26,16 @@ export function prepareErc20EscrowApproval(
   request: ClientTypes.IRequestData,
   paymentTokenAddress: string,
   signerOrProvider: providers.Provider | Signer = getProvider(),
+  amount?: BigNumber,
```

Comment:
as discussed with @olivier7delf , I believe we should stop adding positionnal parameters to payment-processor methods. There should be always the same number of parameters, in the same order, and one of these should be the options, dependant on the PN. 

So typically, `paymentTokenAddress` and `amount` could be grouped in the same param.

Reply:
Yes I agree. we'll need to make a refacto about this at some point. I kind of did it a while back but with the new features we kind of diverged from it.